mscore/tourhandler.cpp

```diff
@@ -0,0 +1,456 @@
+      : QWidget{parent}
+      {
+      widgets = widgetList;
+      newParent();
+      }
+
+//---------------------------------------------------------
+//   newParent
+//---------------------------------------------------------
+
+void OverlayWidget::newParent()
+      {
+      if (!parent())
+            return;
+      parent()->installEventFilter(this);
+      resize(qobject_cast<QWidget*>(parent())->size());
+      raise();
+      }
+
+//---------------------------------------------------------
+//   eventFilter
+//---------------------------------------------------------
+
+bool OverlayWidget::eventFilter(QObject* obj, QEvent* ev)
+      {
+      if (obj == parent()) {
+            if (ev->type() == QEvent::Resize)
+                  resize(static_cast<QResizeEvent*>(ev)->size());
+            else if (ev->type() == QEvent::ChildAdded)
+                  raise();
+            }
+      return QWidget::eventFilter(obj, ev);
+      }
+
+//---------------------------------------------------------
+//   event
+//---------------------------------------------------------
+
+bool OverlayWidget::event(QEvent* ev)
+      {
+      if (ev->type() == QEvent::ParentAboutToChange) {
+            if (parent()) parent()->removeEventFilter(this);
+            }
+      else if (ev->type() == QEvent::ParentChange)
+            newParent();
+      return QWidget::event(ev);
+      }
+
+//---------------------------------------------------------
+//   paintEvent
+//---------------------------------------------------------
+
+void OverlayWidget::paintEvent(QPaintEvent *)
+      {
+      QPainterPath painterPath = QPainterPath();
+      QPainter p(this);
+      QWidget* parentWindow = qobject_cast<QWidget*>(parent());
+      if (parentWindow)
+            painterPath.addRect(parentWindow->rect());
+
+      QPainterPath subPath = QPainterPath();
+      for (QWidget* w : widgets) {
+            if (w->isVisible()) {
+                  // Add widget and children visible region mapped to the parentWindow
+                  QRegion region = w->visibleRegion() + w->childrenRegion();
+                  region.translate(w->mapTo(parentWindow, QPoint(0, 0)));
+                  subPath.addRegion(region);
+                  }
+            }
+      painterPath -= subPath;
+
+      QColor overlayColor = QApplication::palette().color(QPalette::Shadow);
+      overlayColor.setAlpha(128);
+      p.fillPath(painterPath, overlayColor);
+      }
+
+//---------------------------------------------------------
+//   loadTours
+//---------------------------------------------------------
+
+void TourHandler::loadTours()
+      {
+      QStringList nameFilters;
+      nameFilters << "*.tour";
+      QString path = mscoreGlobalShare + "tours";
+
+      QDir dir(path);
+      QStringList pl = dir.entryList(nameFilters, QDir::Files, QDir::Name);
+
+      for (const QString& entry : pl) {
+            QFile* tourFile = new QFile(path + "/" + entry);
```

Comment:
Recently some issues were reported regarding how temporary filenames were being generated for the new extension facility - something about how the path separators were used was breaking things on Linux.  I didn't totally follow that, but please verify this is the recommended way to generate pathnames.

Reply:
The only other way I've been seeing (more often than the one I used here) is to use QDirIterator. I'll use that and also use QDirIterator::Subdirectories so they can be further sorted.

Reply:
No, the only way we separate directories in paths is "/" symbol. https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules#Pathnames

Reply:
OK, thanks for clarifying.  I knew I had seen something about this, didn't realize the bottom line was that we changed *from* the QDir method to just using "/"